policies/templates/gcp_iam_audit_log.yaml

```diff
@@ -0,0 +1,92 @@
+# Copyright 2019 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# GCPIAMAuditLogConstraintV1 verifies that audig logging is enabled in the
+# given service with the given log types.
+
+apiVersion: templates.gatekeeper.sh/v1alpha1
+kind: ConstraintTemplate
+metadata:
+  name: gcp-iam-audit-log-v1
+spec:
+  crd:
+    spec:
+      names:
+        kind: GCPIAMAuditLogConstraintV1
+        plural: gcpiamauditlogconstraintsv1
+      validation:
+        openAPIV3Schema:
+          properties:
+            service:
+              description: "The service that should have audit logging enabled. 
+                Example: 'storage.googleapis.com' for Cloud Storage.
+                'allServices' is a special value that covers all services"
+              type: string
+            log_types:
+              description: "The log types that should be enabled. The valudes are numerical.
```

Comment:
Do you think we should create the mapping underneath and let user enter the actual log type in string instead? That way is less confusing for users.

Reply:
Done. PTAL.